### PR TITLE
Update EstimateGasLimit config name to EstimateLimit

### PR DESCRIPTION
**Pull request by amit-momin**
Updated the `EstimateGasLimit` config name to `EstimateLimit` since this config is under the `GasEstimator`. `Gas` is implied by the config section so there is no need to mention it in the config name.

### CODE REVIEW

**Review by huangzhen1997 — DISMISSED** [review]

File: core/chains/evm/config/toml/config.go [hunk 1]
```diff
@@ -559,7 +559,7 @@ type GasEstimator struct {
 	LimitMultiplier  *decimal.Decimal
 	LimitTransfer    *uint64
 	LimitJobType     GasLimitJobType `toml:",omitempty"`
-	EstimateGasLimit *bool
+	EstimateLimit *bool
```
Comment (dimriou) [inline core/chains/evm/config/toml/config.go]: lint

File: core/chains/evm/txmgr/test_helpers.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func (g *TestGasEstimatorConfig) LimitTransfer() uint64      { return 42 }
 func (g *TestGasEstimatorConfig) PriceMax() *assets.Wei      { return assets.NewWeiI(42) }
 func (g *TestGasEstimatorConfig) PriceMin() *assets.Wei      { return assets.NewWeiI(42) }
 func (g *TestGasEstimatorConfig) Mode() string               { return "FixedPrice" }
-func (g *TestGasEstimatorConfig) EstimateGasLimit() bool     { return false }
+func (g *TestGasEstimatorConfig) EstimateLimit() bool     { return false }
```
Comment (dimriou) [inline core/chains/evm/txmgr/test_helpers.go]: lint

File: core/services/chainlink/config_test.go [hunk 3]
```diff
@@ -531,7 +531,7 @@ func TestConfig_Marshal(t *testing.T) {
 					LimitMax:           ptr[uint64](17),
 					LimitMultiplier:    mustDecimal("1.234"),
 					LimitTransfer:      ptr[uint64](100),
-					EstimateGasLimit:   ptr(false),
+					EstimateLimit:   ptr(false),
```
Comment (dimriou) [inline core/services/chainlink/config_test.go]: lint

**Review by dimriou — DISMISSED** [review]

File: .changeset/loud-windows-call.md [hunk 4]
```diff
@@ -2,4 +2,4 @@
 "chainlink": minor
 ---
 
-Added gas limit estimation feature to EVM gas estimators #added
+Added gas limit estimation feature to EVM gas estimators. Introduced a new config EstimateLimit to toggle this feature. #added
```
Comment (jmank88) [inline .changeset/loud-windows-call.md]:
```suggestion
Added gas limit estimation feature to EVM gas estimators. Introduced a new config `EVM.GasEstimator.EstimateLimit` to toggle this feature. #added
```

**Review by dimriou — APPROVED** [review]

**Review by huangzhen1997 — APPROVED** [review]

**Pull request merged**